From 14d8078ce6454c20c071f2a932899e4e133b92da Mon Sep 17 00:00:00 2001
From: Chris Atkinson <cwa@pipeline.com>
Date: Thu, 1 Jan 2015 16:35:34 -0500
Subject: [PATCH] man: clarify path escaping and reference systemd-escape

This patch adds more detail to the description of how path escaping
operates and provides a pointer to the systemd-escape program. Either
would serve to answer the question raised in the bug report, so
hopefully this will allow it to be closed.

https://bugs.freedesktop.org/show_bug.cgi?id=87688
(cherry picked from commit 995084c7b4683190404ec90f8924b3433b5a1a18)
---
 man/systemd.unit.xml | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/man/systemd.unit.xml b/man/systemd.unit.xml
index f1867ae831..e20851f6c9 100644
--- a/man/systemd.unit.xml
+++ b/man/systemd.unit.xml
@@ -225,12 +225,17 @@
                 the file system namespace. If this applies, a special
                 way to escape the path name is used, so that the
                 result is usable as part of a filename. Basically,
-                given a path, "/" is replaced by "-", and all
-                unprintable characters and the "-" are replaced by
-                C-style "\x2d" escapes. The root directory "/" is
-                encoded as single dash, while otherwise the initial
-                and ending "/" is removed from all paths during
-                transformation. This escaping is reversible.</para>
+                given a path, "/" is replaced by "-" and all other
+                characters which are not ASCII alphanumerics are
+                replaced by C-style "\x2d" escapes (except that "_"
+                is never replaced and "." is only replaced when it
+                would be the first character in the escaped path).
+                The root directory "/" is encoded as single dash,
+                while otherwise the initial and ending "/" are removed
+                from all paths during transformation. This escaping
+                is reversible. Properly escaped paths can be generated
+                using the <citerefentry><refentrytitle>systemd-escape</refentrytitle><manvolnum>1</manvolnum></citerefentry>
+                command.</para>
 
                 <para>Optionally, units may be instantiated from a
                 template file at runtime. This allows creation of
